File: ATOMSwiftUI/mobile/ios/LilithApp/Models/Message.swift
import Foundation

struct Message: Codable, Identifiable {
    let id: String
    let sender_id: String
    let receiver_id: String
    let content: String
    let created_at: String
    let read_at: String?
}

struct ThreadResponse: Codable {
    let items: [Message]
}

File: ATOMSwiftUI/backend/schemas/message.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class MessageSendPayload(BaseModel):
    receiver_id: str
    content: str


class MessageOut(BaseModel):
    id: str
    sender_id: str
    receiver_id: str
    content: str
    created_at: datetime
    read_at: datetime | None = None

    class Config:
        from_attributes = True
